chiselFrontend/src/main/scala/chisel3/core/Mem.scala

```diff
@@ -36,27 +36,30 @@ sealed abstract class MemBase[T <: Data](t: T, val length: Int) extends HasId wi
   /** Creates a read accessor into the memory with static addressing. See the
     * class documentation of the memory for more detailed information.
     */
-  def apply(idx: Int): T = {
+  def apply(idx: Int)/*(implicit compileOptions: CompileOptions)*/: T = {
+    // We can't force this method to take an implicit parameter, since its signature comes from collection.IndexedSeq
+    // How do we thread the caller's CompileOptions through here?
+    import chisel3.core.ExplicitCompileOptions.NotStrict
```

Comment:
What's the correct way to do this? Do we need to make this `apply` method a macro so we can materialize the client's `compileOptions` and pass them along?